sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -40,23 +40,33 @@ object CostBasedJoinReorder extends Rule[LogicalPlan] with PredicateHelper {
     if (!conf.cboEnabled || !conf.joinReorderEnabled) {
       plan
     } else {
+      // Use a map to track the hints on the join items. If a join relation turns out unchanged
```

Comment:
how to define "unchanged"? If `(a join b) join c` becomes `(b join a) join c`, is there any hit we want to retain?